ruby/ql/lib/codeql/ruby/dataflow/BarrierGuards.qll

```diff
@@ -116,7 +116,7 @@ private predicate stringConstArrayInclusionCall(
       isArrayConstant(t.getContainerNode().asExpr(), arr)
     |
       forall(ExprCfgNode elem | elem = arr.getAnArgument() |
-        elem instanceof ExprNodes::StringLiteralCfgNode
+        elem.getConstantValue().isStringlikeValue(_)
```

Comment:
Consider replacing all occurrences of `ExprNodes::StringLiteralCfgNode` in this file.

Reply:
This seems to make sense, I think it would allow us to cover guards like:
```
guardvar = "safe"
case input
  when guardvar
    # input is validated
  else
    # input is not validated
end
```